addons/mail/static/src/components/web_client_view_attachment_view_container/web_client_view_attachment_view_container.js

```diff
@@ -0,0 +1,88 @@
+/** @odoo-module **/
+
+import { useModels } from '@mail/component_hooks/use_models';
+// ensure components are registered beforehand.
+import '@mail/components/web_client_view_attachment_view/web_client_view_attachment_view';
+import { replace } from '@mail/model/model_field_command';
+import { getMessagingComponent } from "@mail/utils/messaging_component";
+
+const { Component, onWillDestroy, onWillUpdateProps } = owl;
+
+const getNextId = (function () {
+    let tmpId = 0;
+    return () => {
+        tmpId += 1;
+        return tmpId;
+    };
+})();
+
+/**
+ * This component abstracts attachment view component to its parent, so that it
+ * can be mounted and receive messaging data even when an attachment view
+ * component cannot be created. Indeed, in order to create an attachment view
+ * component, we must create an attachment view record, the latter requiring
+ * messaging to be initialized. The view may attempt to create an attachment
+ * view before messaging has been initialized, so this component delays the
+ * mounting of attachment view until it becomes initialized.
+ */
+export class WebClientViewAttachmentViewContainer extends Component {
+
+    /**
+     * @override
+     */
+    setup() {
+        useModels();
+        super.setup();
+        this.webClientViewAttachmentView = undefined;
+        this.webClientViewAttachmentViewId = getNextId();
+        this.insertFromProps(this.props);
+        onWillUpdateProps(nextProps => this.insertFromProps(nextProps));
+        onWillDestroy(() => this.deleteRecord());
+    }
+
+    deleteRecord() {
+        if (this.webClientViewAttachmentView) {
+            if (this.webClientViewAttachmentView.exists()) {
+                this.webClientViewAttachmentView.delete();
+            }
+            this.webClientViewAttachmentView = undefined;
+        }
+    }
+
+    /**
+     * @private
+     */
+    async insertFromProps(props) {
```

Comment:
why not prefixing it with _ if it's supposed to be private?

Reply:
I'll make it private, but private method in JS is an obsolete context, the difference is not very important.